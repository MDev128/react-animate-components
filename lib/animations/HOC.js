"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var HOC = function HOC(componentName, animationName) {
  var _class, _temp2;

  return _temp2 = _class = function (_Component) {
    _inherits(_class, _Component);

    function _class() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, _class);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = _class.__proto__ || Object.getPrototypeOf(_class)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
        styles: {}
      }, _this.componentDidMount = function () {
        var _this$props = _this.props,
            duration = _this$props.duration,
            timingFunction = _this$props.timingFunction,
            delay = _this$props.delay,
            direction = _this$props.direction,
            iterations = _this$props.iterations,
            backfaceVisible = _this$props.backfaceVisible,
            fillMode = _this$props.fillMode;


        _this.setState({
          styles: {
            display: "inline-block" /** <span> elements can't be animated with CSS. In order to animate them with 
                                    CSS we assign display property to them */
            , // Works but will be updated later.
            animation: animationName + " " + duration + " " + timingFunction + " " + delay + " " + fillMode + " " + direction + " " + iterations,
            backfaceVisibility: "" + backfaceVisible
          }
        });
      }, _temp), _possibleConstructorReturn(_this, _ret);
    }

    _createClass(_class, [{
      key: "render",


      // Using <span> instead of composedComponent because each component has a base component <span> only.
      value: function render() {
        return _react2.default.createElement(
          "span",
          { style: this.state.styles },
          this.props.children
        );
      }
    }]);

    return _class;
  }(_react.Component), _class.defaultProps = {
    duration: "1s",
    timingFunction: "ease",
    delay: "0s",
    direction: "normal",
    iterations: "1",
    backfaceVisible: "visible",
    fillMode: "none"
  }, _temp2;
};

exports.default = HOC;