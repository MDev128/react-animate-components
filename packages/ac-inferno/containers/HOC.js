// @flow

import Inferno from 'inferno';
import Component from 'inferno-component';
import { Render, derive } from '../utils/index';

type DefaultProps = {
  duration: string,
  timingFunction: string,
  delay: string,
  direction: string,
  iterations: string,
  backfaceVisible: string,
  fillMode: string,
  playState: string,
  as: string,
  component?: Function,
  children?: Element<any>,
  forceInterpolate?: Object,
  style?: Object
}

type Props = {
  duration: string,
  timingFunction: string,
  delay: string,
  direction: string,
  iterations: string,
  backfaceVisible: string,
  fillMode: string,
  playState: string,
  as: string,
  component?: Function,
  children?: Element<any>,
  forceInterpolate?: Object,
  style?: Object
}

type State = {
  styles: Object
}

function hoc(ComposedComponent: string, AnimationName: string): Function {
  class _Animation extends Component<DefaultProps, Props, State> {
    static displayName = `${ComposedComponent}`;

    static defaultProps = {
      duration: '1s',
      timingFunction: 'ease',
      delay: '0s',
      direction: 'normal',
      iterations: '1',
      backfaceVisible: 'visible',
      fillMode: 'none',
      playState: 'running',
      as: 'div'
    };

    state = {
      styles: {}
    };

    componentDidMount = () => {
      this.setAnimation(this.props);
    };

    componentWillReceiveProps = (nextProps: Props) => {
      // Interpolation of new animation properties
      const deriveInterpolationFromNextProps = derive(nextProps, AnimationName);

      // Old interpolation string
      const deriveInterpolationFromPrevProps = derive(
        this.props,
        AnimationName
      );

      if (
        deriveInterpolationFromNextProps !== deriveInterpolationFromPrevProps
      ) {
        this.setState({
          styles: Object.assign(
            {
              animation: `${deriveInterpolationFromNextProps}`,
              backfaceVisibility: `${nextProps.backfaceVisible}`
            },
            this.props.style || {}
          )
        });
      }
    };

    setAnimation = (props: Props) => {
      // Keyframes Interpolation and Force Interpolation
      const deriveInterpolation = derive(props, AnimationName);

      this.setState({
        styles: Object.assign(
          {
            animation: `${deriveInterpolation}`,
            backfaceVisibility: `${props.backfaceVisible}`
          },
          this.props.style || {}
        )
      });
    };

    render (): ?React$Element<any> {
      const {
        children,
        duration,
        timingFunction,
        delay,
        direction,
        iterations,
        backfaceVisible,
        fillMode,
        playState,
        forceInterpolate,
        as,
        style,
        component,
        ...rest
      } = this.props;

      return Render(ComposedComponent, this.props, this.state, rest, ComposedComponent);
    }
  }

  return _Animation;
}

export default hoc;
