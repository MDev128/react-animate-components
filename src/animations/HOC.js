// @flow

import React, { Component } from 'react';

type Props = {
  duration: string,
  timingFunction: string,
  delay: string,
  direction: string,
  iterations: string,
  backfaceVisible: string,
  fillMode: string,
  playState: string
};

type DefaultProps = {
  duration: string,
  timingFunction: string,
  delay: string,
  direction: string,
  iterations: string,
  backfaceVisible: string,
  fillMode: string,
  playState: string
};

type State = {
  styles: Object
};

let HOC = (componentName: string, animationName: string) =>
  class extends Component {
    state = {
      styles: {}
    };

    static defaultProps = {
      duration: "1s",
      timingFunction: "ease",
      delay: "0s",
      direction: "normal",
      iterations: "1",
      backfaceVisible: "visible",
      fillMode: "none",
      playState: "running"
    };

    componentDidMount = () => {
      const {
        duration,
        timingFunction,
        delay,
        direction,
        iterations,
        backfaceVisible,
        fillMode,
        playState
      } = this.props;

      this.setState({
        styles: {
          display: "inline-block" /** <span> elements can't be animated with CSS. In order to animate them with 
                                CSS we assign display property to them */,
          // Works but will be updated later.
          animation: `${animationName} ${duration} ${timingFunction} ${delay} ${iterations} ${direction} ${fillMode} ${playState}`,
          backfaceVisibility: `${backfaceVisible}`
        }
      });
    }

    // Using <span> instead of composedComponent because each component has a base component <span> only.
    render() {
      return (
        <span style={this.state.styles}>
          {this.props.children}
        </span>
      );
    }
  };

export default HOC;