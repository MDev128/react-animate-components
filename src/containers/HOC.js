// @flow

import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import checkTag from "html-tags";

import { attributes, shouldNotBeUndefined } from "../utils/attributes";

import getElementType from "../mods/getElementType";
import avoidNest from "../mods/avoidNesting";

import derive from "../utils/state";
import { direction, interpolateValidators } from "../utils/keyValidators";

type Props = {
  duration: string,
  timingFunction: string,
  delay: string,
  direction: string,
  iterations: string,
  backfaceVisible: string,
  fillMode: string,
  playState: string,
  forceInterpolate: Object,
  children: Object,
  as: string,
  style: Object,
  component: Function
};

type DefaultProps = {
  duration: string,
  timingFunction: string,
  delay: string,
  direction: string,
  iterations: string,
  backfaceVisible: string,
  fillMode: string,
  playState: string,
  as: string
};

type State = {
  styles: Object
};

function setTypes(ComposedComponent) {
  return {
    direction: PropTypes.oneOf([
      "normal",
      "reverse",
      "alternate",
      "alternate-reverse",
      "initial",
      "inherit",
    ]),
    fillMode: PropTypes.oneOf(["none", "forwards", "backwards", "both"]),
    playState: PropTypes.oneOf(["paused", "running"]),
    timingFunction: PropTypes.oneOf([
      "linear",
      "ease",
      "ease-in",
      "ease-out",
      "ease-in-out",
      "step-start",
      "step-end",
    ]),
    backfaceVisible: PropTypes.oneOf(["visible", "hidden"]),
    /* eslint-disable object-shorthand */
    /* eslint-disable func-names */
    as: function (props, propName) {
      const prop = props[propName];
      const err = `Warning: '${prop}' passed to '${ComposedComponent}' component is not a valid html tag.`;
      /* eslint-disable no-console */
      return checkTag.includes(prop) ? null : console.error(err);
    },
    forceInterpolate: PropTypes.objectOf((propValue, key) => {
      direction(key, propValue);
      interpolateValidators(key);
    }),
    component: PropTypes.func,
  };
}

const defaultProps = {
  duration: "1s",
  timingFunction: "ease",
  delay: "0s",
  direction: "normal",
  iterations: "1",
  backfaceVisible: "visible",
  fillMode: "none",
  playState: "running",
  as: "div",
};

function HOC(ComposedComponent: string, AnimationName: string) {
  class _Animation extends PureComponent<DefaultProps, Props, State> {
    static displayName = `${ComposedComponent}`;

    static defaultProps = {
      duration: "1s",
      timingFunction: "ease",
      delay: "0s",
      direction: "normal",
      iterations: "1",
      backfaceVisible: "visible",
      fillMode: "none",
      playState: "running",
      as: "div",
    };

    state = {
      styles: {},
    };

    componentDidMount = () => {
      this.store(this.props);
    };

    componentWillReceiveProps = (nextProps: Props) => {
      // Interpolation of new animation properties
      const deriveInterpolationFromNextProps = derive(nextProps, AnimationName);

      // Old interpolation string
      const deriveInterpolationFromPrevProps = derive(
        this.props,
        AnimationName,
      );

      if (
        deriveInterpolationFromNextProps !== deriveInterpolationFromPrevProps
      ) {
        this.setState({
          styles: Object.assign(
            {
              animation: `${deriveInterpolationFromNextProps}`,
              backfaceVisibility: `${nextProps.backfaceVisible}`,
            },
            this.props.style || {},
          ),
        });
      }
    };

    store = (props: Props) => {
      // Keyframes Interpolation and Force Interpolation
      const deriveInterpolation = derive(props, AnimationName);

      this.setState({
        styles: Object.assign(
          {
            animation: `${deriveInterpolation}`,
            backfaceVisibility: `${props.backfaceVisible}`,
          },
          this.props.style || {},
        ),
      });
    };

    render(): ?React$Element<any> {
      const ElementType = getElementType(ComposedComponent, this.props);

      const { styles } = this.state;
      const { children } = this.props;

      // Alternate, pass a component as a prop to an animation component
      const Wrapper = this.props.component;

      // Validates the DOM nesting of elements.
      const NormalizedComponent = avoidNest(ElementType, children);

      // Add rest of the props except component props(html-elements)
      const reactHtmlAttributes = attributes(this.props);

      return (
        <NormalizedComponent
          style={styles}
          {...shouldNotBeUndefined(reactHtmlAttributes)}
        >
          { Wrapper ? React.createElement(Wrapper, children) : children }
        </NormalizedComponent>
      );
    }
  }

  _Animation.propTypes = setTypes(ComposedComponent);
  _Animation.defaultProps = defaultProps;

  return _Animation;
}

export default HOC;
