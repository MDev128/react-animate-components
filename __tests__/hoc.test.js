import React from "react";
import renderer from "react-test-renderer";
import { keyframes } from "styled-components";
import "jest-styled-components";
import { shallow } from "enzyme";

import HOC from "../src/containers/HOC";

const sample = keyframes`
	0% {
		opacity: 0;
	}

	100% {
		opacity: 1;
	}
`;

const Sample = HOC("Sample", sample);

const sampleComponent = (
  <Sample duration="2s">
    Hello World!
  </Sample>
);

let App = () => {
	return (
		<div>
			<p>Sign of Times</p>
		</div>
	);
};

describe('High Order Component', () => {
	it("should be a function", () => {
	  expect(typeof HOC).toBe("function");
	});

	it("wraps the children under h1 element", () => {
	  const tree = renderer
	    .create(
	      <Sample duration="2s" as="h1">
	        Hello World!
	      </Sample>
	    )
	    .toJSON();
	  expect(tree).toMatchSnapshot();
	});

	it("renders the Component passed through component prop", () => {
		const tree = renderer.create(
			<Sample duration="2s" as="h1" component={App} />
		).toJSON();

		expect(tree).toMatchSnapshot();
	})

	it("wraps the children under div (default)", () => {
	  const tree = renderer.create(sampleComponent).toJSON();
	  expect(tree).toMatchSnapshot();
	});

	it("matches the keyframes created by styled-components and default props", () => {
	  const tree = renderer.create(sampleComponent).toJSON();

	  expect(tree).toMatchStyledComponentsSnapshot();
	});

	it("calls componentDidMount lifecycle method", () => {
	  const wrapper = shallow(sampleComponent);
	  wrapper.instance().componentDidMount();
		// Also calls store() method when component mounts.
	});

	it("updates the styles when the component mounts", () => {
	  const wrapper = shallow(sampleComponent);
	  wrapper.instance().componentDidMount();

	  expect(wrapper.state("styles")).toEqual({
	    animation: "UNeSH 2s ease 0s 1 normal none running",
	    backfaceVisibility: "visible"
	  });
	});
})
