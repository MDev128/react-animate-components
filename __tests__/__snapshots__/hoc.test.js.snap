// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`High Order Component matches the keyframes created by styled-components and default props 1`] = `


<div
  accept={undefined}
  acceptCharset={undefined}
  accessKey={undefined}
  action={undefined}
  allowFullScreen={undefined}
  allowTransparency={undefined}
  alt={undefined}
  async={undefined}
  autoComplete={undefined}
  autoFocus={undefined}
  autoPlay={undefined}
  capture={undefined}
  cellPadding={undefined}
  cellSpacing={undefined}
  challenge={undefined}
  charSet={undefined}
  checked={undefined}
  cite={undefined}
  classID={undefined}
  className={undefined}
  colSpan={undefined}
  cols={undefined}
  content={undefined}
  contentEditable={undefined}
  contextMenu={undefined}
  controls={undefined}
  coords={undefined}
  crossOrigin={undefined}
  data={undefined}
  dateTime={undefined}
  default={undefined}
  defer={undefined}
  dir={undefined}
  disabled={undefined}
  download={undefined}
  draggable={undefined}
  encType={undefined}
  form={undefined}
  formAction={undefined}
  formEncType={undefined}
  formMethod={undefined}
  formNoValidate={undefined}
  formTarget={undefined}
  frameBorder={undefined}
  headers={undefined}
  height={undefined}
  hidden={undefined}
  high={undefined}
  href={undefined}
  hrefLang={undefined}
  htmlFor={undefined}
  httpEquiv={undefined}
  icon={undefined}
  id={undefined}
  inputMode={undefined}
  integrity={undefined}
  is={undefined}
  keyParams={undefined}
  keyType={undefined}
  kind={undefined}
  label={undefined}
  lang={undefined}
  list={undefined}
  loop={undefined}
  low={undefined}
  manifest={undefined}
  marginHeight={undefined}
  marginWidth={undefined}
  max={undefined}
  maxLength={undefined}
  media={undefined}
  mediaGroup={undefined}
  method={undefined}
  min={undefined}
  minLength={undefined}
  multiple={undefined}
  muted={undefined}
  name={undefined}
  noValidate={undefined}
  nonce={undefined}
  open={undefined}
  optimum={undefined}
  pattern={undefined}
  placeholder={undefined}
  poster={undefined}
  preload={undefined}
  profile={undefined}
  radioGroup={undefined}
  readOnly={undefined}
  rel={undefined}
  required={undefined}
  reversed={undefined}
  role={undefined}
  rowSpan={undefined}
  rows={undefined}
  sandbox={undefined}
  scope={undefined}
  scoped={undefined}
  scrolling={undefined}
  seamless={undefined}
  selected={undefined}
  shape={undefined}
  size={undefined}
  sizes={undefined}
  span={undefined}
  spellCheck={undefined}
  src={undefined}
  srcDoc={undefined}
  srcLang={undefined}
  srcSet={undefined}
  start={undefined}
  step={undefined}
  style={
    Object {
      "animation": "UNeSH 2s ease 0s 1 normal none running",
      "backfaceVisibility": "visible",
    }
  }
  summary={undefined}
  tabIndex={undefined}
  target={undefined}
  title={undefined}
  type={undefined}
  useMap={undefined}
  value={undefined}
  width={undefined}
  wmode={undefined}
  wrap={undefined}
>
  Hello World!
</div>
`;

exports[`High Order Component renders the Component passed through component prop 1`] = `


<h1
  accept={undefined}
  acceptCharset={undefined}
  accessKey={undefined}
  action={undefined}
  allowFullScreen={undefined}
  allowTransparency={undefined}
  alt={undefined}
  async={undefined}
  autoComplete={undefined}
  autoFocus={undefined}
  autoPlay={undefined}
  capture={undefined}
  cellPadding={undefined}
  cellSpacing={undefined}
  challenge={undefined}
  charSet={undefined}
  checked={undefined}
  cite={undefined}
  classID={undefined}
  className={undefined}
  colSpan={undefined}
  cols={undefined}
  content={undefined}
  contentEditable={undefined}
  contextMenu={undefined}
  controls={undefined}
  coords={undefined}
  crossOrigin={undefined}
  data={undefined}
  dateTime={undefined}
  default={undefined}
  defer={undefined}
  dir={undefined}
  disabled={undefined}
  download={undefined}
  draggable={undefined}
  encType={undefined}
  form={undefined}
  formAction={undefined}
  formEncType={undefined}
  formMethod={undefined}
  formNoValidate={undefined}
  formTarget={undefined}
  frameBorder={undefined}
  headers={undefined}
  height={undefined}
  hidden={undefined}
  high={undefined}
  href={undefined}
  hrefLang={undefined}
  htmlFor={undefined}
  httpEquiv={undefined}
  icon={undefined}
  id={undefined}
  inputMode={undefined}
  integrity={undefined}
  is={undefined}
  keyParams={undefined}
  keyType={undefined}
  kind={undefined}
  label={undefined}
  lang={undefined}
  list={undefined}
  loop={undefined}
  low={undefined}
  manifest={undefined}
  marginHeight={undefined}
  marginWidth={undefined}
  max={undefined}
  maxLength={undefined}
  media={undefined}
  mediaGroup={undefined}
  method={undefined}
  min={undefined}
  minLength={undefined}
  multiple={undefined}
  muted={undefined}
  name={undefined}
  noValidate={undefined}
  nonce={undefined}
  open={undefined}
  optimum={undefined}
  pattern={undefined}
  placeholder={undefined}
  poster={undefined}
  preload={undefined}
  profile={undefined}
  radioGroup={undefined}
  readOnly={undefined}
  rel={undefined}
  required={undefined}
  reversed={undefined}
  role={undefined}
  rowSpan={undefined}
  rows={undefined}
  sandbox={undefined}
  scope={undefined}
  scoped={undefined}
  scrolling={undefined}
  seamless={undefined}
  selected={undefined}
  shape={undefined}
  size={undefined}
  sizes={undefined}
  span={undefined}
  spellCheck={undefined}
  src={undefined}
  srcDoc={undefined}
  srcLang={undefined}
  srcSet={undefined}
  start={undefined}
  step={undefined}
  style={
    Object {
      "animation": "UNeSH 2s ease 0s 1 normal none running",
      "backfaceVisibility": "visible",
    }
  }
  summary={undefined}
  tabIndex={undefined}
  target={undefined}
  title={undefined}
  type={undefined}
  useMap={undefined}
  value={undefined}
  width={undefined}
  wmode={undefined}
  wrap={undefined}
>
  <div>
    <p>
      Sign of Times
    </p>
  </div>
</h1>
`;

exports[`High Order Component wraps the children under div (default) 1`] = `


<div
  accept={undefined}
  acceptCharset={undefined}
  accessKey={undefined}
  action={undefined}
  allowFullScreen={undefined}
  allowTransparency={undefined}
  alt={undefined}
  async={undefined}
  autoComplete={undefined}
  autoFocus={undefined}
  autoPlay={undefined}
  capture={undefined}
  cellPadding={undefined}
  cellSpacing={undefined}
  challenge={undefined}
  charSet={undefined}
  checked={undefined}
  cite={undefined}
  classID={undefined}
  className={undefined}
  colSpan={undefined}
  cols={undefined}
  content={undefined}
  contentEditable={undefined}
  contextMenu={undefined}
  controls={undefined}
  coords={undefined}
  crossOrigin={undefined}
  data={undefined}
  dateTime={undefined}
  default={undefined}
  defer={undefined}
  dir={undefined}
  disabled={undefined}
  download={undefined}
  draggable={undefined}
  encType={undefined}
  form={undefined}
  formAction={undefined}
  formEncType={undefined}
  formMethod={undefined}
  formNoValidate={undefined}
  formTarget={undefined}
  frameBorder={undefined}
  headers={undefined}
  height={undefined}
  hidden={undefined}
  high={undefined}
  href={undefined}
  hrefLang={undefined}
  htmlFor={undefined}
  httpEquiv={undefined}
  icon={undefined}
  id={undefined}
  inputMode={undefined}
  integrity={undefined}
  is={undefined}
  keyParams={undefined}
  keyType={undefined}
  kind={undefined}
  label={undefined}
  lang={undefined}
  list={undefined}
  loop={undefined}
  low={undefined}
  manifest={undefined}
  marginHeight={undefined}
  marginWidth={undefined}
  max={undefined}
  maxLength={undefined}
  media={undefined}
  mediaGroup={undefined}
  method={undefined}
  min={undefined}
  minLength={undefined}
  multiple={undefined}
  muted={undefined}
  name={undefined}
  noValidate={undefined}
  nonce={undefined}
  open={undefined}
  optimum={undefined}
  pattern={undefined}
  placeholder={undefined}
  poster={undefined}
  preload={undefined}
  profile={undefined}
  radioGroup={undefined}
  readOnly={undefined}
  rel={undefined}
  required={undefined}
  reversed={undefined}
  role={undefined}
  rowSpan={undefined}
  rows={undefined}
  sandbox={undefined}
  scope={undefined}
  scoped={undefined}
  scrolling={undefined}
  seamless={undefined}
  selected={undefined}
  shape={undefined}
  size={undefined}
  sizes={undefined}
  span={undefined}
  spellCheck={undefined}
  src={undefined}
  srcDoc={undefined}
  srcLang={undefined}
  srcSet={undefined}
  start={undefined}
  step={undefined}
  style={
    Object {
      "animation": "UNeSH 2s ease 0s 1 normal none running",
      "backfaceVisibility": "visible",
    }
  }
  summary={undefined}
  tabIndex={undefined}
  target={undefined}
  title={undefined}
  type={undefined}
  useMap={undefined}
  value={undefined}
  width={undefined}
  wmode={undefined}
  wrap={undefined}
>
  Hello World!
</div>
`;

exports[`High Order Component wraps the children under h1 element 1`] = `


<h1
  accept={undefined}
  acceptCharset={undefined}
  accessKey={undefined}
  action={undefined}
  allowFullScreen={undefined}
  allowTransparency={undefined}
  alt={undefined}
  async={undefined}
  autoComplete={undefined}
  autoFocus={undefined}
  autoPlay={undefined}
  capture={undefined}
  cellPadding={undefined}
  cellSpacing={undefined}
  challenge={undefined}
  charSet={undefined}
  checked={undefined}
  cite={undefined}
  classID={undefined}
  className={undefined}
  colSpan={undefined}
  cols={undefined}
  content={undefined}
  contentEditable={undefined}
  contextMenu={undefined}
  controls={undefined}
  coords={undefined}
  crossOrigin={undefined}
  data={undefined}
  dateTime={undefined}
  default={undefined}
  defer={undefined}
  dir={undefined}
  disabled={undefined}
  download={undefined}
  draggable={undefined}
  encType={undefined}
  form={undefined}
  formAction={undefined}
  formEncType={undefined}
  formMethod={undefined}
  formNoValidate={undefined}
  formTarget={undefined}
  frameBorder={undefined}
  headers={undefined}
  height={undefined}
  hidden={undefined}
  high={undefined}
  href={undefined}
  hrefLang={undefined}
  htmlFor={undefined}
  httpEquiv={undefined}
  icon={undefined}
  id={undefined}
  inputMode={undefined}
  integrity={undefined}
  is={undefined}
  keyParams={undefined}
  keyType={undefined}
  kind={undefined}
  label={undefined}
  lang={undefined}
  list={undefined}
  loop={undefined}
  low={undefined}
  manifest={undefined}
  marginHeight={undefined}
  marginWidth={undefined}
  max={undefined}
  maxLength={undefined}
  media={undefined}
  mediaGroup={undefined}
  method={undefined}
  min={undefined}
  minLength={undefined}
  multiple={undefined}
  muted={undefined}
  name={undefined}
  noValidate={undefined}
  nonce={undefined}
  open={undefined}
  optimum={undefined}
  pattern={undefined}
  placeholder={undefined}
  poster={undefined}
  preload={undefined}
  profile={undefined}
  radioGroup={undefined}
  readOnly={undefined}
  rel={undefined}
  required={undefined}
  reversed={undefined}
  role={undefined}
  rowSpan={undefined}
  rows={undefined}
  sandbox={undefined}
  scope={undefined}
  scoped={undefined}
  scrolling={undefined}
  seamless={undefined}
  selected={undefined}
  shape={undefined}
  size={undefined}
  sizes={undefined}
  span={undefined}
  spellCheck={undefined}
  src={undefined}
  srcDoc={undefined}
  srcLang={undefined}
  srcSet={undefined}
  start={undefined}
  step={undefined}
  style={
    Object {
      "animation": "UNeSH 2s ease 0s 1 normal none running",
      "backfaceVisibility": "visible",
    }
  }
  summary={undefined}
  tabIndex={undefined}
  target={undefined}
  title={undefined}
  type={undefined}
  useMap={undefined}
  value={undefined}
  width={undefined}
  wmode={undefined}
  wrap={undefined}
>
  Hello World!
</h1>
`;
